hooks/boot/21-upgrade_storage_engine.rb

```diff
@@ -0,0 +1,6 @@
+app_option(
+  '--upgrade-storage-engine',
```

Comment:
I'd opt for `mongo-storage-engine` to make it clear that this is not some generic storage engine concept.